### DISCUSSION TITLE
Problems when using external fonts

### DISCUSSION THREAD

**Discussion by SashaCX75** [zepp-health/zeppos-forum#147] [Developer Feedback]
I'm trying to use an external font in the **TEXT** widget by pointing to it with the **font:** property. But this works very badly. Depending on the text displayed, some of the characters may not show up, or the entire caption may disappear.

**Comment by MyoungXUE** [comment]
Hello, could you please provide the font file and project code snippet to help us troubleshoot the problem.

**Comment by SashaCX75** [comment]
[test 4.zip](https://github.com/zepp-health/zeppos-forum/files/11326124/test.4.zip)
Here, in the "CeraStencilPRO-Bold" font, some of the characters are not displayed. 

Also, if I make a watch face in Visual Studio Code, the font files are not added to the compiled file.

**Comment by SashaCX75** [comment]
[test 4.zip](https://github.com/zepp-health/zeppos-forum/files/11326382/test.4.zip)
The same fonts are not displayed at all when the size is increased.

**Comment by SashaCX75** [comment]
It seems that only the characters that were present when the text was first displayed are displayed. The other characters are skipped. Perhaps the characters are cached the first time they are displayed. The character cache is only updated after reloading the watch.
